crates/pathfinder/src/rpc/types.rs

```diff
@@ -630,11 +649,17 @@ pub mod reply {
 
     impl From<sequencer::reply::transaction::Transaction> for Transaction {
         fn from(txn: sequencer::reply::transaction::Transaction) -> Self {
+            Self::from(&txn)
+        }
+    }
+
+    impl From<&sequencer::reply::transaction::Transaction> for Transaction {
+        fn from(txn: &sequencer::reply::transaction::Transaction) -> Self {
```

Comment:
Why via reference? Wouldn't it be better to drop the clone if txn is consumed, and use into_iter instead of iter?
This would be possible if you added a consuming `MaybeBlock::as_[transactions|receipts](self)` to the enum api.

Reply:
This is only really used in the [from_sequencer_scoped](https://github.com/eqlabs/pathfinder/pull/441/files#diff-de073c53108c3c0e6543a72a3893ca2c6978b212062c620a4fb4ba2cccf882ddR369). The issue is that it cannot consume `self` because it requires `self` afterwards still.

Alternatively, one can just clone the transaction as well. However, adding a consuming API for the transactions isn't useful as we almost always need more than the transactions.

Reply:
Thanks for linking the very line